packages/tpetra/core/src/Tpetra_Details_unpackCrsGraphAndCombine_def.hpp

```diff
@@ -911,42 +878,89 @@ unpackCrsGraphAndCombine(
         importLIDs.getRawPtr(), importLIDs.size(),
         true, "import_lids");
 
-  auto local_graph = sourceGraph.getLocalGraph();
-  auto local_col_map = sourceGraph.getColMap()->getLocalMap();
+  // mfh Could fix the use of protected data by adding a method
+  // to CrsGraph that returns the "local, globally indexed graph".
+  // However, this function needs to change those arrays anyway,
+  // so we might as well make this function a friend of CrsGraph.
+
+  //FIXME (tjf 11 Oct 2018) need to use non-protected data
+  indices_type indices("indices", graph.k_gblInds1D_.extent(0));
+  Kokkos::deep_copy(indices, graph.k_gblInds1D_);
+
+  row_ptrs_type row_ptrs_beg("row_ptrs_beg", graph.k_rowPtrs_.extent(0));
+  Kokkos::deep_copy(row_ptrs_beg, graph.k_rowPtrs_);
+
+  const auto N = (row_ptrs_beg.extent(0) == 0 ? 0 : row_ptrs_beg.extent(0) - 1); // TODO fix auto
+  row_ptrs_type row_ptrs_end("row_ptrs_end", N);
+
+  bool refill_num_row_entries = false;
+  if (graph.k_numRowEntries_.extent(0) > 0) {
+    // Case 1: Packed storage
+    refill_num_row_entries = true;
+    auto num_row_entries = graph.k_numRowEntries_;
+    Kokkos::parallel_for("Fill end row pointers", range_policy(0, N),
+        KOKKOS_LAMBDA(const size_t i){
+          row_ptrs_end(i) = row_ptrs_beg(i) + num_row_entries(i);
+      });
+
+  } else {
+    // mfh If packed storage, don't need row_ptrs_end to be separate allocation;
+    // could just have it alias row_ptrs_beg+1.
```

Comment:
Are we leaving this comment for later?

Reply:
Yes, this function will change once the new interface is implemented